### PR TITLE
Merge @ImagicalCorp changes into official repo.

### PR DESCRIPTION
**Pull request by keithkfng**
I know a lot of people are just staring blankly at this PR now, and I know why. When I first created the @ImagicalCorp repo, I never expected it to be this big. But since the community and a lot of people requested @ImagicalCorp's commits to be merged into the official repo, I've decided to go for it with @PEMapModder's help. Whoever is merging this PR - I'd advise you to create a duplicate of the original mcpe-0.13 branch before merging. I can't guarantee the 100% no bugs guarantee to you today.

People/teams I'd like to thank on behalf of the community:
@PocketMine team - for continuing the mainstream development of this project
@ImagicalCorp team - for doing so much work on the unofficial repo
@shoghicp - for coding the main structure of PocketMine.
@PEMapModder - for helping @ImagicalCorp with git issues, providing support when the team was having trouble AND provided suggestions.

### CODE REVIEW

**Comment by PEMapModder** [comment]
Why do you need to give credit to @shoghicp? Nothing in this pull request was written by him.

**Comment by keithkfng** [comment]
True... But he wrote the original code XD

**Comment by PEMapModder** [comment]
But you are adding changes to original code. You aren't posting anything written from the original repo, only additions/amendments.

Also please read the edited comment.

To prevent further chat, please PM me on Gitter.

**Comment by GODLIKE5** [comment]
How do you guys make a phar a zip which will be used as a Pocketmine build... Like to add stuff to it.. like mcpe 0.13.0 beta builds for pocketmine.

File: src/pocketmine/Player.php [hunk 1]
```diff
@@ -113,7 +115,11 @@
 use pocketmine\network\protocol\PlayerActionPacket;
 use pocketmine\network\protocol\PlayStatusPacket;
 use pocketmine\network\protocol\RespawnPacket;
+<<<<<<< HEAD
```
Comment (SOF3) [inline src/pocketmine/Player.php]: Irony: the third line changed is already wrong.

**Pull request closed without merging**